app/parse.py

```diff
@@ -8,9 +19,137 @@ class Quote:
     tags: list[str]
 
 
+@dataclass
+class Author:
+    name: str
+    biography: str
+
+
+QUOTE_FIELDS = [field.name for field in fields(Quote)]
+AUTHOR_FIELDS = [field.name for field in fields(Author)]
+authors_list = []
+
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="[%(levelname)8s]: %(message)s",
+    handlers=[
+        logging.FileHandler("parser.log"),
+        logging.StreamHandler(sys.stdout),
+    ]
+)
+
+
+def parse_single_quote(quote_soup: Tag) -> Quote:
+    author_name = quote_soup.select_one(".author").text
+    if author_name not in authors_list:
+        authors_list.append(author_name)
+    return Quote(
+        text=quote_soup.select_one(".text").text,
+        author=author_name,
+        tags=quote_soup.select_one(".tags").text.split()[1:]
+    )
+
+
+def get_single_author(name: str, biography: str) -> Author:
+    return Author(
+        name=name,
+        biography=biography,
+    )
+
+
+def get_single_page_quotes(page_soup: Tag) -> [Quote]:
+    quotes = page_soup.select(".quote")
+    return [parse_single_quote(quote_soup) for quote_soup in quotes]
+
+
+def get_all_pages_quotes() -> [Quote]:
+    # logging
+    time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    logging.info(f"{time} - Start parsing quotes")
+
+    page_text = requests.get(BASE_URL).content
+    first_page_soup = BeautifulSoup(page_text, "html.parser")
+
+    all_quotes = get_single_page_quotes(first_page_soup)
+
+    # next button searching
+    next_element = first_page_soup.select_one(".next")
+    next_page_number = next_element.a["href"].split("/")[-2]
```

Comment:
The `next_page_number` extraction assumes a specific URL structure. If the URL structure changes, this could break. Consider using a more robust method to extract the page number.